### ISSUE TITLE
Relationships between Grappler, GraphOptimizer and GraphOptimizationPass

### ISSUE THREAD

**Issue by wangsiyu** [tensorflow/tensorflow#23553]
There are so many graph optimization tools in TensorFlow Runtime such as `Grappler`, `GraphOptimizer` and `GraphOptimizationPass`. This makes me confused.  Is there any plan to unify these interfaces？Currently, it seems that these optimizers could not replace with each other.

`Grappler` is very light weight because it does graph optimization on `GraphDef` .The limitation is that  these `optimizers` can only run after `Placer.run()`.  However,  some optimizers in `Grappler` will rewrite placement information. Therefore,  this will possibly cause some placement errors without the protection of `Placer`(Is it reasonable to  run `Placer` after running grappler?). Another, the `FeedBack` function is not being used in framework.

`GraphOptimizationPass` does optimization on `Graph` object. The advantage is that these optimizers can be specified with execution stage(PRE_PLACEMENT, POST_PLACEMENT......) .This is necessary. I think the execution stage of graph optimizers should be treated differently. Some optimizers should be done before placement and the other may be suitable for post partitioning and etc. 

As for `GraphOptimizer`, I heard that it will not be updated in the future. 

It will be nice to unify them and bring their strength together.  @caisq

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sachinprasadhs** [comment]
Could you please refer [this](https://www.tensorflow.org/guide/graph_optimization) Graph Optimization documentation in the recent Tensorflow version and compare the different Graph Optimizer, If you still feel the issue is not resolved feel free to open a new issue.
